### DISCUSSION TITLE
Access Alloy database from WebView script

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13627] [Q&A] [answered]
The index.xml contains a webView with reference to /assets/test/webview.html. The webview inserts a javascript file from same directory. Is it possible to access Alloy database from this js file?

views/index.xml:
```
...
<WebView id="testWebView" url="/test/webview.html" />
...
```
assets/test/webview.html
```
...
<script type="text/javascript" src="test.js"></script>
...

```
assets/test/test.js

```
//Ti.Database.open("_alloy_");

```

**Comment by m1ga** [comment, marked as answer]
They live in different folders. You can try this PR https://github.com/tidev/titanium_mobile/pull/13229 to access a different folder or simply copy it inside your test folder at runtime. You still would need to use a sqlite library to work with it in your page.

I would read it in Ti and pass in the data (doing that for charts). If you do more work: just create a normal Ti layout instead of a webview

**Reply by beppo-ivel** [reply]
>I would read it in Ti and pass in the data (doing that for charts)

Can't find it anymore, but once there was a tutorial to plot a d3 chart in titanium. That's why I'm using the webView as I described above. Passing data in general is not the problem. But if I start the app I don't know how to wait for till the webview and dependencies are loaded?

>If you do more work: just create a normal Ti layout instead of a webview

You mean copy the d3.js library in to the lib folder and create the graph in a standard view?

**Reply by m1ga** [reply]
no, for a chart I wouldn't go native! Didn't know if you were just showing some lists of data or so.

I'm using chartist but other Slack users reported that they use highchart, chartjs or different libraries. d3 should be fine too. The normal `load` event is fine and inside there I'll fire
```js
Ti.App.fireEvent('app:showGraph', {
	data: JSON.stringify(statisticData),
	year: currentYear,
	dark: isDark
});
``` 

If you need to initialize stuff in the webview that is ready after DOM ready you could use a event inside from webview -> Ti to let Ti know when you are ready. Using a Ti setTimeout and checking a webview variable with evaljs is another thing I've done in the past.

https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_How-tos/Integrating_Web_Content/Communication_Between_WebViews_and_Titanium.html
